core/win32/os.c

```diff
@@ -2497,6 +2497,15 @@ static bool
 os_take_over_thread(dcontext_t *dcontext, HANDLE hthread, thread_id_t tid, bool suspended)
 {
     bool success = true;
+
+    if (DYNAMO_OPTION(skip_terminating_threads)) {
+        if (nt_is_thread_terminating(hthread)) {
```

Comment:
I wonder if we can avoid the extra call to NtSuspendThread here by having nt_thread_suspend return the NTSTATUS -- but it gets more complicated if we want to support `suspended==true` so maybe simplest to leave it how you have it.

Reply:
My logic was "break as little as possible", but if you think we should, then I'll take a look at it.

Reply:
Just leave it.